packages/trilinoscouplings/examples/scaling/example_Poisson_stk.cpp

```diff
@@ -922,27 +890,35 @@ int main(int argc, char *argv[]) {
   }
 
   /**********************************************************************************/
-  /************************ ADJUST MATRIX AND RHS FOR BCs ***************************/
+  /************************** DIRICHLET BC SETUP ************************************/
   /**********************************************************************************/
+  Epetra_Vector lhsVector(globalMapG,true);
 
-  // Apply stiffness matrix to v
-  Epetra_MultiVector rhsDir(globalMapG,true);
-  StiffMatrix.Apply(v,rhsDir);
-
-  // Update right-hand side
-  rhsVector.Update(-1.0,rhsDir,1.0);
-
-  // Loop over local boundary nodes and replace rhs values with boundary values
-  for (size_t i = 0; i < bcNodeVec.size(); i++) {
-
-    int lid = bcNodeVec[i];
-    rhsVector[0][lid]=v[0][lid];
-
+  std::vector<int> ownedBoundaryNodes; ownedBoundaryNodes.reserve(bcNodes.size());
+  // Loop over boundary nodes
+  for (unsigned i = 0; i < bcNodes.size(); i++) {
+    int bcNodeId = bulkData.identifier(bcNodes[i]);
+    int lid = globalMapG.LID((int) bcNodeId -1);
+    if(lid != -1) {
+      ownedBoundaryNodes.push_back(lid);
+      
+      // get coordinates for this node
+      entity_type bcnode = bulkData.get_entity(NODE_RANK,bcNodeId);
```

Comment:
It looks like 'bcnode' is the same as 'bcNodes[i]' above, so you don't need this bulkdata.get_entity by Global-ID, which is a map find.